pkg/segment/query/processor/evalcommand.go

```diff
@@ -18,22 +18,130 @@
 package processor
 
 import (
+	"fmt"
+	"io"
+
 	"github.com/siglens/siglens/pkg/segment/query/iqr"
 	"github.com/siglens/siglens/pkg/segment/structs"
+	"github.com/siglens/siglens/pkg/segment/utils"
+	toputils "github.com/siglens/siglens/pkg/utils"
 )
 
 type evalProcessor struct {
 	options *structs.EvalExpr
 }
 
+func evaluateValueExpr(valueExpr *structs.ValueExpr, fieldToValue map[string]utils.CValueEnclosure) (interface{}, error) {
+	switch valueExpr.ValueExprMode {
+	case structs.VEMConditionExpr:
+		value, err := valueExpr.ConditionExpr.EvaluateCondition(fieldToValue)
+		if err != nil {
+			return nil, fmt.Errorf("evaluateValueExpr: failed to evaluate condition expr, err: %v", err)
+		}
+		return value, nil
+	case structs.VEMStringExpr:
+		value, err := valueExpr.EvaluateValueExprAsString(fieldToValue)
+		if err != nil {
+			return nil, fmt.Errorf("evaluateValueExpr: failed to evaluate string expr, err: %v", err)
+		}
+		return value, nil
+	case structs.VEMNumericExpr:
+		value, err := valueExpr.EvaluateToFloat(fieldToValue)
+		if err != nil {
+			// It failed to evaluate to a float, it could possibly that the field given is a string
+			valueStr, err := valueExpr.EvaluateToString(fieldToValue)
+			if err != nil {
+				return nil, fmt.Errorf("evaluateValueExpr: failed to evaluate numeric expr to numeric and string expr, err: %v", err)
+			}
+
+			return valueStr, err
+		}
+		return value, nil
+	case structs.VEMBooleanExpr:
+		value, err := valueExpr.EvaluateToString(fieldToValue)
+		if err != nil {
+			return nil, fmt.Errorf("evaluateValueExpr: failed to evaluate boolean expr, err: %v", err)
+		}
+		return value, nil
+	case structs.VEMMultiValueExpr:
+		mvSlice, err := valueExpr.EvaluateToMultiValue(fieldToValue)
+		if err != nil {
+			return nil, fmt.Errorf("evaluateValueExpr: failed to evaluate multi value expr, err: %v", err)
+		}
+		return mvSlice, nil
+	default:
+		return nil, fmt.Errorf("evaluateValueExpr: unknown value expr mode %v", valueExpr.ValueExprMode)
+	}
+}
+
 func (p *evalProcessor) Process(iqr *iqr.IQR) (*iqr.IQR, error) {
-	panic("not implemented")
+	if iqr == nil {
+		return nil, io.EOF
+	}
+	if p.options == nil {
+		return iqr, fmt.Errorf("evalProcessor.Process: options is nil")
+	}
+	if p.options.ValueExpr == nil {
+		return iqr, toputils.TeeErrorf("evalProcessor.Process: ValueExpr is nil")
+	}
+	fieldsInExpr := p.options.ValueExpr.GetFields()
+	allCnames, err := iqr.GetColumns()
+	if err != nil {
+		return nil, fmt.Errorf("evalProcessor.Process: failed to get all column names, err: %v", err)
+	}
+	if len(fieldsInExpr) == 1 && fieldsInExpr[0] == "*" {
+		fieldsInExpr = []string{}
+		for cname := range allCnames {
+			fieldsInExpr = append(fieldsInExpr, cname)
+		}
+	}
```

Comment:
I think allCnames is only used inside the `if`, so you should move `allCnames, err := iqr.GetColumns()` inside the `if` as well, so we don't do that computation when we don't need it

Reply:
addressed in the latest commit.